src/Spice86.Core/Emulator/OperatingSystem/LocalFileSearchManager.cs

```diff
@@ -0,0 +1,134 @@
+            return false;
+        }
+
+        return WildFileCmp(filename.AsSpan(), pattern.AsSpan());
+    }
+
+    private static bool WildFileCmp(ReadOnlySpan<char> sourceFilename, ReadOnlySpan<char> pattern) {
+        if (sourceFilename.Length > 0 && pattern.Length == 0) {
+            return false;
+        }
+
+        if (pattern.Length > LfnNamelength) {
+            return false;
+        }
+
+        // Fast path: exact case-insensitive match (covers common no-wildcard cases)
+        if (sourceFilename.Equals(pattern, StringComparison.OrdinalIgnoreCase)) {
+            return true;
+        }
+
+        // Skip “hidden” dot-files if the pattern uses wildcards (except "." / "..")
+        if (WildcardMatchesHiddenFile(sourceFilename, pattern)) {
+            return false;
+        }
+        
+        // Uppercase once into fixed-size 8.3 stacks with space padding
+        Span<char> fileName = stackalloc char[DosMfnlength];
+        Span<char> fileExt = stackalloc char[DosExtlength];
+        Span<char> wildName = stackalloc char[DosMfnlength];
+        // wild ext needs an extra slot to check the 4th char like DOSBox
+        Span<char> wildExt = stackalloc char[DosExtlength + 1];
+
+        SplitTo83(sourceFilename, fileName, fileExt, out _);
+        SplitTo83(pattern, wildName, wildExt, out int wildExtLength);
+
+        // ---- NAME compare (no early '*' accept; '*' just ends name matching) ----
+        if (CompareSegment(fileName, wildName, DosMfnlength, false) == false) {
+            return false;
+        }
+        // the original code tolerated extra chars in pattern name beyond 8 unless it's a non-'*' check byte; our fixed-size truncation mirrors that behavior
+
+        // ---- EXT compare (early '*' accept) ----
+        return CompareSegment(fileExt, wildExt, DosExtlength, true) switch {
+            true => true,
+            false => false,
+            // If wild ext has a 4th char, and it's not '*', reject (DOSBox-like behavior)
+            _ => wildExtLength <= DosExtlength || wildExt[DosExtlength] == '*'
+        };
+    }
+
+    // Common 8.3 segment compare:
+    // - Compares up to 'length' characters.
+    // - '?' matches any char (including space padding).
+    // - If '*' is encountered:
+    //     - when earlyStarAcceptsTrue == true (extension), return true immediately;
+    //     - otherwise (name), treat as "stop comparing here" and return null (no decision).
+    // Returns:
+    //   true => definitively accept (only possible with earlyStarAcceptsTrue and '*' seen)
+    //   false => mismatch
+    //   null => matched the segment fully (or name-stopped at '*'), no final decision
+    private static bool? CompareSegment(ReadOnlySpan<char> target, ReadOnlySpan<char> pattern, int length,
+        bool earlyStarAcceptsTrue) {
+        for (int i = 0; i < length; i++) {
+            char wc = pattern[i];
+            if (wc == '*') {
+                return earlyStarAcceptsTrue ? true : null;
```

Comment:
What is the meaining of 'earlyStarAcceptsTrue' ? Isn't there a better name for it ?

This dosbox source code always confused me - same for the DOSBox Staging authors.